chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/statistics/NonconformityTypeStatisticsDAOImpl.java

```diff
@@ -0,0 +1,73 @@
+package gov.healthit.chpl.dao.statistics;
+
+import gov.healthit.chpl.auth.Util;
+import gov.healthit.chpl.dao.impl.BaseDAOImpl;
+import gov.healthit.chpl.domain.DateRange;
+import gov.healthit.chpl.dto.NonconformityTypeStatisticsDTO;
+import gov.healthit.chpl.entity.surveillance.NonconformityTypeStatisticsEntity;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.persistence.Query;
+
+import org.springframework.stereotype.Repository;
+
+@Repository("nonconformityTypeStatisticsDAO")
+public class NonconformityTypeStatisticsDAOImpl extends BaseDAOImpl implements NonconformityTypeStatisticsDAO {
+
+	@Override
+	public List<NonconformityTypeStatisticsDTO> getAllNonconformityStatistics(
+			DateRange dateRange) {
+		String hql = "FROM NonconformityTypeStatisticsEntity WHERE";
+		
+		if(dateRange == null) {
+            hql += " deleted = false";
+		} else {
+			hql += "(deleted = false AND creationDate <= :endDate) "
+                + " OR "
+                + "(deleted = true AND creationDate <= :endDate AND lastModifiedDate > :endDate)";
```

Comment:
If line 38 is false, because the creationDate is greater than :endDate, then line 30 can never be true.

Also, all branches of this if chunk have "deleted = false". That should be taken out of the conditional sections entirely.

Why use a "dateRange" but only use the end date of that range?

Why use a dateRange at all? The only mention the ticket makes of dates is on one end because we can only have data from when CHPL started, and on the other as our way of getting out of needing to do real-time stats.

Line 35 should be inside the else block from line 27, because you're checking dateRange == null twice for no particular reason

Reply:
I think checking twice is necessary because the first time is to build the HQL string that makes the query object (and you can't make a query object without it), then once you have the query object you need to set the parameters on it so you have to check the same conditions again to know which parameters to set since the query is being built dynamically and can vary.

Reply:
Regarding moving the deleted = false, I misunderstood at first but now agree... since it's in both of the conditional sections it could/should be moved out of those and added to the hql before the if/else checks.